### ISSUE TITLE
Update cell in IPython notebook with Stan progress (warmup, sampling)

### ISSUE THREAD

**Issue by ariddell** [stan-dev/pystan#153]
Right now IPython notebook ignores stdout and stderr. Perhaps PyStan can even render some sort of progress widget if it detects it is in an IPython notebook environment.

**Comment by d10genes** [comment]
pymc3 has a nice progressbar [here](https://github.com/pymc-devs/pymc3/blob/e482d0f41a345e5a74557fe1e23a342d1124f06b/pymc3/progressbar.py) that updates in the notebook periodically.


    In [1]: import pymc3 as pm
    In [2]: prog = pm.progressbar.progress_bar(30)
    In [3]: prog.update(5)
     [--------         23%                  ] 7 of 30 complete in 1.2 sec

**Comment by ariddell** [comment]
That's great. Once the callback refactor (stan-dev/stan/issues/1361) is done, this should be straightforward. (Right now PyStan doesn't get any information about the sampler state.)

**Comment by ihincks** [comment]
In the interim, I found a nice package which collects C-level stdout/stderr: https://github.com/minrk/wurlitzer

However, it only returns all of the text after the sampling has run, so not useful for progress display. I asked about it here: https://github.com/minrk/wurlitzer/issues/9

**Comment by ariddell** [comment]
Thanks! This problem bothers me as well.

It's fixed in PyStan 3. Code should be public soon.

**Comment by ihincks** [comment]
Oh, cool, thanks for responding. Is it being developed in a private repo or will it be the next major release of this repo?

**Comment by ariddell** [comment]
It will appear in this repo eventually and an alpha version will be available soon. (It would be great to have some testers!) You might want to watch https://github.com/stan-dev/pystan/issues/305 for updates.

**Comment by ariddell** [comment]
Fixed in PyStan 3

**Comment by cmelog** [comment]
Hi I am using the last version of pystan but I do not see any output on the console while the sampling is running. I have included verbose = True in the model. I hope you can provide some feedback.

**Comment by ariddell** [comment]
Would you mind opening a new issue? This is about IPython/Jupyter notebooks.
